pyblish_lite/control.py

```diff
@@ -37,15 +34,26 @@ class Controller(QtCore.QObject):
     # Emitted when processing has finished
     was_finished = QtCore.Signal()
 
+    PART_COLLECT = 'collect'
+    PART_VALIDATE = 'validate'
+    PART_EXTRACT = 'extract'
+    PART_CONFORM = 'conform'
+
     def __init__(self, parent=None):
         super(Controller, self).__init__(parent)
 
-        self.context = list()
-        self.plugins = list()
+        self.context = pyblish.api.Context()
+        self.plugins = {}
 
         # Data internal to the GUI itself
         self.is_running = False
 
+        # Bools to know in which part of publishing is right now
+        # TODO may be stored in `processing` dict?
+        self.validated = False
```

Comment:
It shouldn't be necessary to keep track of these. You should only need to know at what order you are. E.g. `0.5` or `0.78` etc. It's important not to hard-code these CVEI steps, but to keep plug-ins in a linear order, in the order they are defined. CVEI is merely a suggestion. If you need to determine order, parse the current order, e.g. `order_to_cvei(1.5) == "validation"`. But really, you shouldn't have to. Is this for visualisation? If so, it shouldn't be in the controller.

Reply:
This is for validation also connected with issue above... Don't want to change - would require a lot of changes and I'm not sure if result would work properly.

Reply:
This applies to the comment above as well.

If I remember correctly all of these were added to make sure we're able to stop after each section if something goes wrong. Shouldn't that be the case? 

Aren't we intrinsically expecting CVEI workflow in this GUI? I mean if we label things `collect`, `validate`, `extract` and `integrate` in the UI, then it should be ok to refer to them in the code as well shouldn't it? The naming of the 4 steps goes all the way back to pyblish-base, so there is a built-in assumption anyway to some degree.

Reply:
> Aren't we intrinsically expecting CVEI workflow in this GUI?

We shouldn't do this, because stopping at validation is just what the default registered test does. It's optional. We need to stop whenever the registered test tells us to, so that users custom tests remain useful.

> If I remember correctly all of these were added to make sure we're able to stop after each section if something goes wrong. Shouldn't that be the case?

There's only one reason to stop, and that's when the registered test says so.

At the end of the day, every plug-in has an order that falls between some start and end number. Default is 0-4, but users can assign any order. Including 1-100, or -500 to 50,001, with a custom test to stop things at every 1,000th order or whenever you reach 0.

Plug-ins are sorted by this order, and run in that order. A test is called for each instance/plug-in pair to evaluate whether or not to stop. The GUI just needs to run plug-ins in that order, and respect that test. Anything else is optional visual flare, like CVEI and those labels.

Reply:
Right I see. We'll have a look. cheers